extends Node2D

signal change_stage(next)
signal start_conversation(lines, item)

var controller: Node = null
@export
var yuhina: Node = null:
	set(value):
		$God.yuhina = value

const yuhina_start_at: Vector2 = Vector2(70, 90)

var yuhina_and_god = [
	"[b][color=blue]ç¥æ§˜[/color][/b] ã¾ãŸä¼šã£ãŸãª",
	"[b][color=red]yuhina[/color][/b] ã‚â€¼ï¸ [color=red]é€”ä¸­çœç•¥[/color]",
	"[b][color=blue]ç¥æ§˜[/color][/b] ä»Šã®ãŠå‰ã«ã¯ç„¡ç†ã˜ã‚ƒãªğŸ‘»",
	"[b][color=red]yuhina[/color][/b] ã§ã¯ã©ã†ã—ãŸã‚‰ï¼ï¼Ÿ",
	"[b][color=blue]ç¥æ§˜[/color][/b] ãŠå‰ã«ä¼èª¬ã®ã‚¢ã‚¤ãƒ†ãƒ ã‚’æ¸¡ãã†ã€ãã‚Œï¼ğŸ",
]

# 2å›ç›®ä»¥é™ã®ã‚»ãƒªãƒ•
var yuhina_and_god2 = [
	"[b][color=blue]ç¥æ§˜[/color][/b] ä¼èª¬ã®ã‚¢ã‚¤ãƒ†ãƒ ã€Œè¸Šã‚‹å°å­¦4å¹´ç”Ÿäººå½¢ğŸ§¸ã€ã‚’å¤§äº‹ã«ã—ã¦ãŠã‚‹ã‹ã®",
	"[b][color=blue]ç¥æ§˜[/color][/b] ã‚¤ãƒ™ãƒ³ãƒˆã‚’ã‚¯ãƒªã‚¢ã—ã¦ã‹ã‚‰ã¾ãŸæ¥ã‚‹ãŒã‚ˆã„ã€‚"
]

func _on_god_yuhina_in(first):
	if first:
		start_conversation.emit(yuhina_and_god, 2)
	else:
		start_conversation.emit(yuhina_and_god2, 0)

func _on_kitty_kitty_meets_yuhina():
	start_conversation.emit(['ğŸ± ã‚¦ãƒã«æˆ»ã‚‹ãƒ‹ãƒ£ãƒ¼'], false)
	change_stage.emit(1)
